src/variable.c

```diff
@@ -1132,11 +1131,17 @@ mrb_class_find_path(mrb_state *mrb, struct RClass *c)
 mrb_int
 mrb_obj_iv_tbl_memsize(mrb_state* mrb, mrb_value obj)
 {
-  return sizeof(iv_tbl) +
-    (sizeof(segment) * ceil(iv_size(mrb, mrb_obj_ptr(obj)->iv)/
-                            MRB_IV_SEGMENT_SIZE));
+  size_t ivsize = iv_size(mrb, mrb_obj_ptr(obj)->iv);
+  size_t ivsegs = (ivsize + MRB_IV_SEGMENT_SIZE - 1) / MRB_IV_SEGMENT_SIZE;
+  return sizeof(iv_tbl) + (sizeof(segment) * ivsegs);
```

Comment:
This change in `mrb_obj_iv_tbl_memsize` is great. It should remain.

Reply:
It seems that the memory size cannot be calculated by this way because deleted instance variable slots remain.

I have a feeling that deleted slots will disappear if `iv_del` does the followings:

* Move the last instance variable to the deleted slot.
* Free the last segment if it is empty.

Reply:
I tried to address `iv_tbl` shrink but found out that removing instance variables is very rare. So I concluded it's not worth the complexity. Maybe we will reimplement the `iv_tbl` in the future (to reduce memory). We will revisit removing `iv` then.